vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -180,35 +198,46 @@ void TouchEventHandler::OnPointerConcluded(TouchEventType eventType, const winrt
 
   // if the view has a Tag, update the pointer info.
   // Regardless of that, ensure we Dispatch & cleanup the pointer
-  int64_t tag;
+  std::vector<int64_t> tagsForBranch;
   xaml::UIElement sourceElement(nullptr);
-  if (TagFromOriginalSource(args, &tag, &sourceElement))
+  const auto kind = GetPointerEventKind(eventType);
+  const auto reactArgs = winrt::make<winrt::Microsoft::ReactNative::implementation::ReactPointerEventArgs>(kind, args);
+  const auto argsImpl =
+      winrt::get_self<winrt::Microsoft::ReactNative::implementation::ReactPointerEventArgs>(reactArgs);
+  if (PropagatePointerEventAndFindReactTarget(reactArgs, &tagsForBranch, &sourceElement))
     UpdateReactPointer(m_pointers[*optPointerIndex], args, sourceElement);
 
   if (m_pointers[*optPointerIndex].isLeftButton) {
-    DispatchTouchEvent(eventType, *optPointerIndex);
+    if (argsImpl->CaptureReleased()) {
+      // Force cancellation if ReleaseCapture is called ReactPointerEventArgs
+      DispatchTouchEvent(TouchEventType::Cancel, *optPointerIndex);
+    } else if (!argsImpl->DefaultPrevented()) {
```

Comment:
We should change this to send cancel on either CaptureReleased or DefaultPrevented, so we don't get lingering events in RN

Reply:
Likely only when not preventing capture lost events